python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -49,6 +50,53 @@
 )
 from .partition_key_range import PartitionKeyRange
 
+POST_DST_TRANSITION_SUFFIX = "_POST_DST_TRANSITION"
+
+
+def dst_safe_strftime(dt: datetime, timezone_str: str, fmt: str) -> str:
+    """A method for converting a datetime to a string which will append a suffix in cases where
+    the resulting timestamp would be ambiguous due to DST transitions.
+    """
+    time_str = dt.strftime(fmt)
+    try:
+        create_pendulum_time(
+            dt.year,
+            dt.month,
+            dt.day,
+            dt.hour,
+            dt.minute,
+            dt.second,
+            dt.microsecond,
+            tz=timezone_str,
+            dst_rule=pendulum.TRANSITION_ERROR,
+        )
+    except pendulum.tz.exceptions.AmbiguousTime:  # type: ignore
+        if dt.fold == 1:
+            return time_str + POST_DST_TRANSITION_SUFFIX
```

Comment:
alternate implementation that i'm not sure is better or worse

```

def is_ambiguous_time(dt: datetime):
    # Inspired by https://github.com/sdispater/pendulum/blob/75a87a4e3cab45c5f9c38ca6060a54337dcadd09/src/pendulum/tz/timezone.py#L107-L132
    tz = check.not_none(dt.tzinfo)
    offset_before = cast(
        timedelta,
        (tz.utcoffset(dt.replace(fold=0)) if dt.fold else tz.utcoffset(dt)),
    )
    offset_after = cast(
        timedelta,
        (tz.utcoffset(dt) if dt.fold else tz.utcoffset(dt.replace(fold=1))),
    )
    return offset_before > offset_after


def dst_safe_strftime(dt: datetime, fmt: str) -> str:
    """A method for converting a datetime to a string which will append a suffix in cases where
    the resulting timestamp would be ambiguous due to DST transitions.
    """
    time_str = dt.strftime(fmt)
    if not _IS_PENDULUM_2:
        # fold attribute not set correctly in pendulum 1
        return time_str

    if is_ambiguous_time(dt):
        if dt.fold:
            return time_str + POST_DST_TRANSITION_SUFFIX
    return time_str
```

Reply:
I like this a bit more -- it's more explicit and allows for us to put fancy logic in is_ambiguous_time if we wish